sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/TestContext.java

```diff
@@ -0,0 +1,56 @@
+package com.pulumi.deployment.internal;
+
+import com.google.common.collect.ImmutableMap;
+import com.pulumi.Config;
+import com.pulumi.Context;
+import com.pulumi.context.LoggingContext;
+import com.pulumi.core.Output;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class TestContext implements Context {
```

Comment:
what is this class? looks like accidental complexity

Reply:
Not at all accidental, to run callbacks `void init(Context ctx)` in *test* context we need to pass something implementing the `Context` interface. It seemed to me the easiest is to implement TestContext interface, how else would we do this? Are you suggesting using ContextInternal instance here? It looked involved but maybe possible.

Reply:
OK, after working with the code I understand now, it's cool idea to not have to modify too much.
When we have a full runner based on `PulumiInternal` this will not be needed, because we'll be able to supply a real one. IMO this is important for the user to be able to use testing framework almost in the same way as they'd use the normal API.
We might want to extend the `ContextInternal` with a `TestContext` in the future but I haven't found a use case for it yet.

Reply:
Ok sound good, I'll put a bit of effort to removing this, but if it looks too deep, I'll just put a TODO here to remove this and link to an issue. We can at the very least easily make this a private class, not public.